nix/hm-module.nix

```diff
@@ -7,8 +7,7 @@ self: {
   cfg = config.wayland.windowManager.hyprland;
   defaultHyprlandPackage = self.packages.${pkgs.stdenv.hostPlatform.system}.default.override {
     enableXWayland = cfg.xwayland.enable;
-    hidpiXWayland = cfg.xwayland.hidpi;
-    inherit (cfg) nvidiaPatches;
+    inherit (cfg) enableNvidiaPatches;
```

Comment:
This is out of scope for this PR:
I have handled the package overrides a little differently. The way I did it allows the user to set the `package` but also use the `enable*` options that override it.
See: <https://github.com/spikespaz/hyprland-flake/blob/3642290700deb55b097275d1425c63327431dd4c/hm-module/config.nix#L234-L239>

Reply:
I feel the difference is minimal.

Reply:
> I feel the difference is minimal.

True, but it still depends. The only real difference is if someone wants to override the package but still have the module options applied, your way would not work and would mean that the package takes precedence over options. From your perspective though, if they override the package, they probably know what they're doing and would supply those arguments in the override anyway.

I just did this so that you could use a fork or a different branch of Hyprland and have the options applied, assuming the user is not experienced and would be confused that the options stop working.

Reply:
No, I literally mean just the formatting is different. We set `package` and the options get applied to it, and that forms `finalPackage` which we then use everywhere else. It's equivalent to your code.

Reply:
Oh my bad, that's in the NixOS module (and the upstream HM module).

I didn't want to spend any more time working on this, provided we'd be integrating your module soon:tm:.